### PR TITLE
Fix 404 pages

### LINKED ISSUE #745: Our 404 page is unreachable
We should be able to tell the user if they have hit a genuine 404 - right now, we seem to use hang with whitespace.

### PR DESCRIPTION
**Pull request by barrymcgee**
## Done

- Fix 404 pages

## QA

- Pull code
- Run `./run clean && ./run serve`
- Open http://0.0.0.0:8036/fdlkfjdkjf
- Verify 404 page is shown
- Click all other routes and ensure they load as expected

## Details

Fixes #745

### CODE REVIEW

**Comment by anthonydillon** [comment]
Worth getting a design review as it's a new view.

File: src/components/Routes/Routes.js [hunk 1]
```diff
@@ -35,7 +40,20 @@ export function Routes() {
...
+        </Login>
+      </Route>
     );
   });
+
+  return (
+    <Switch>
+      {routes}
+      <Route>
+        <NotFound path="*" />
```
Comment (hatched) [inline src/components/Routes/Routes.js:55]: the `path` prop here is not being used, did you mean to put this on the `Route`?

File: src/pages/NotFound/NotFound.test.js [hunk 2]
```diff
@@ -1,10 +1,24 @@
...
 describe("NotFound page", () => {
-  it("renders without crashing and matches snapshot", () => {
-    const wrapper = shallow(<NotFound />);
-    expect(wrapper).toMatchSnapshot();
+  it("should display when unknown route is accessed", () => {
+    const store = mockStore(dataDump);
+    const wrapper = mount(
+      <Provider store={store}>
+        <MemoryRouter initialEntries={["/foobar11"]}>
+          <NotFound />
```
Comment (hatched) [inline src/pages/NotFound/NotFound.test.js:18]: This test isn't actually testing anything, it's just rendering the `NotFound` component. You'll want to use the `TestRoute` component. You can see it in action in the `InfoPanel` or `ModelDetail` tests.

File: src/components/Routes/Routes.js [hunk 3]
```diff
@@ -35,7 +40,20 @@ export function Routes() {
       );
     }
     return (
-      <Route key={path[0]} path={path[0]} exact component={path[1].component} />
+      <Route key={path[0]} path={path[0]} exact>
```
Comment (hatched) [inline src/components/Routes/Routes.js:43]: This is now wrapping each route in a `Login` component, not all routes. I'm curious why that change was made? I'm not totally sure why this structure change solves the problem, can you elaborate?

**Review by hatched — CHANGES_REQUESTED** [review]

File: src/components/Routes/Routes.js (hunk 1 above)
Comment (barrymcgee) [reply src/components/Routes/Routes.js:55]: I meant to remove it, now removed.

File: src/components/Routes/Routes.js (hunk 3 above)
Comment (barrymcgee) [reply src/components/Routes/Routes.js:43]: A switch statement does not work for nested routes, so I needed to move the <Login> inside each route.

Comment (hatched) [reply src/components/Routes/Routes.js:43]: Ahh makes sense, thanks

**Comment by long-chung** [comment]
Good to go from design. 
*Will look at a redesign of the 404 page at a later date as per conversation we had in standup on 17.11.2020

File: src/pages/NotFound/NotFound.test.js [hunk 4]
```diff
@@ -1,10 +1,24 @@
...
+  it("should display when unknown route is accessed", () => {
+    const store = mockStore(dataDump);
+    const wrapper = mount(
+      <Provider store={store}>
+        <MemoryRouter initialEntries={["/foobar11"]}>
+          <Routes />
+        </MemoryRouter>
+      </Provider>
+    );
+    expect(wrapper.find("NotFound").length).toBe(1);
```
Comment (hatched) [inline src/pages/NotFound/NotFound.test.js:22]: Could you also add a check to make sure nothing else was rendered by `Routes` that wasn't supposed to be?

**Review by hatched — APPROVED** [review]

**Pull request merged**